src/core/federation.ts

```diff
@@ -281,6 +287,125 @@ function isANodeJSCodedError(e: Error): e is NodeJSCodedError {
   return typeof (e as NodeJSCodedError).code === "string";
 }
 
+/**
+ * A network reactor is a reactor handling network actions (NetworkReceiver and NetworkSender).
```

Comment:
Is there any reason to have `NetworkSender` and `NetworkReceiver` have a common base class? It seems to me they should have very little in common.

Reply:
A more concrete question would be: why not define a `NetworkSender` and `NetworkReceiver` class instead of this common `NetworkReactor` class?

Reply:
They should have a common parent class so that we can use them when analyzing dependencies implied by TPO levels. **However**, TPO-level things are not included in this branch, so I'll define a `NetworkSender` and `NetworkReceiver` class instead of the common class.

Reply:
Now, network receiver and sender reactors are instantiated as `NetworkReceiver` and `NetworkSender`, respectively.  For more details, please refer to the commit [e36fb0b](https://github.com/lf-lang/reactor-ts/pull/186/commits/e36fb0b4658e1f469ec37dfcced754e291b9982d).